napari/utils/colormaps/colormaps.py

```diff
@@ -111,6 +117,31 @@ def _low_discrepancy_image(image, seed=0.5, margin=1 / 256):
     return image_out
 
 
+def color_dict_to_colormap(color_dict):
+    """
+    Generate a color map based on the given color dictionary
+    Parameters
+    ----------
+    color_dict : dict of label and color string matches
+
+    Returns
+    -------
+    colormap : Colormap with provided control colors
+    custom_color : dict of label to color control point within colormap
+    """
+    custom_color = {0: 0.0}
```

Comment:
I think the addition of the background transparent color_dict should happen in the labels layer using something like
```python
if self._background_label is not in color_dict:
    color_dict[self._background_label] = '#000000' # transparent color?
```

Reply:
I think also in the labels layer we should have 
```python
if None not in color_dict:
    color_dict[None] = '#000000' #black color?
```
and then inside `custom_color` make `None` get mapped to `1.0` and inside the colormap make `1.0` get mapped to `color_dict[None]`

This way if someone want's to fully customize the coloring using
```python
color_dict = {0: 'white', 1: 'red', 2:'blue', None: 'green'}
```
then the background label would be white, 1, 2 would be red/ blue and all other labels would be green - pretty cool no?

Reply:
For the sake of consistency and uniformity I think it would also be nice to use those color converter functions that @kevinyamauchi mentioned earlier to standardize all the values in the `color_dict` to be 1x4 arrays of RGBA values when the dictionary is passed or ever edited.

i.e.
```python
{1: 'red'} #becomes
{1: np.array([1, 0, 0, 1])}
```
I think having our internal representations of color always be the same will help us in the long run. Note here we don't need to actually make the `Nx4` array. Those color transformation utilities should make this conversion easy. We might want to make a dedicated one to standardize color_dicts that we can import into the labels layer and can be independently unit tested

Reply:
got it, I think I misunderstood some of the comments, the update will be pushed soon